### ISSUE TITLE
EXC_BAD_ACCESS using AFNetworking 2.5 with iOS10

### ISSUE THREAD

**Issue by lauriek334** [AFNetworking/AFNetworking#3730]
My AFNetworking code downloads over 10,000 images from a server in the background - it has been working without issue with ios7,8,9 for the last 12 months. When I run the code under ios10, it runs well for a while and then I get a EXC_BADACCESS  in AFURLConnectionOperation.m in the method: networkRequestThreadEntryPoint on the line [runloop run]; (line 171)
Here's the routine:
```
+ (void)networkRequestThreadEntryPoint:(id)__unused object {
    @autoreleasepool {
        [[NSThread currentThread] setName:@"AFNetworking"];

        NSRunLoop *runLoop = [NSRunLoop currentRunLoop];
        [runLoop addPort:[NSMachPort port] forMode:NSDefaultRunLoopMode];
        [runLoop run];
    }
}
```

![bad_exec_afnetworking](https://cloud.githubusercontent.com/assets/342894/19206823/c05207b2-8cb9-11e6-952d-8fda696241b6.png)

Do I need to update to AFNetworking 3.0 to safely run under iOS10?

**Comment by kcharwood** [comment]
NSURLConnection has been deprecated, so I would recommend updating. We are no longer supporting the codebase that uses NSURLConnection. 🍻

**Comment by vinod1988** [comment]
@lauriek334 is your issue resolved after updating the library v3.0. I have similar kind of issue in my app as well.
